inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -143,6 +150,26 @@ def arxiv_refextract(obj, eng):
         obj.log.error("Not able to download and process the PDF")
 
 
+def arxiv_derive_inspire_categories(obj, eng):
+    """Derive inspire-categories from arxiv-categories
+
+    :param obj: Workflow Object to process
+    :param eng: Workflow Engine processing the object
+    """
+    obj.data.setdefault('inspire_categories', [])
+
+    for arxiv_eprint in obj.data['arxiv_eprints']:
+        for arxiv_category in get_value(arxiv_eprint, 'categories'):
```

Comment:
```python
from itertools import chain

[...]

for arxiv_category in chain.from_iterable(get_value(obj.data, 'arxiv_eprints.categories')):
    [...]
```

That said, this is already used elsewhere, so it should become an helper in `inspirehep/utils/record.py`.